components/suggest/src/query.rs

```diff
@@ -118,4 +120,122 @@ impl SuggestionQuery {
             ..self
         }
     }
+
+    // Other Functionality
+
+    /// Parse the `keyword` field into a set of keywords.
+    ///
+    /// This is used when passing the keywords into an FTS search.  It:
+    ///   - Strips out any `():^*"` chars.  These are typically used for advanced searches, which
+    ///     we don't support and it would be weird to only support for FTS searches, which
+    ///     currently means Fakespot searches.
+    ///   - Splits on whitespace to get a list of individual keywords
+    ///
+    #[cfg(feature = "fakespot")]
+    pub(crate) fn parse_keywords(&self) -> Vec<&str> {
+        self.keyword
+            .split(|c: char| {
+                c.is_whitespace()
+                    || c == '('
+                    || c == ')'
+                    || c == ':'
+                    || c == '^'
+                    || c == '*'
+                    || c == '"'
+            })
+            .filter(|s| !s.is_empty())
+            .collect()
+    }
+
+    /// Create an FTS query term for our keyword(s)
+    #[cfg(feature = "fakespot")]
+    pub(crate) fn fts_query(&self) -> String {
+        // Create a string to hold the formatted query, try to make the capacity large enough so
+        // that we don't need re-allocations
+        let mut fts_query = String::with_capacity(self.keyword.len() * 2);
+        // Quote and append each term from `query`
+        let keywords = self.parse_keywords();
+
+        if keywords.is_empty() {
+            return String::from("\"\"");
+        }
+        for keyword in keywords[..keywords.len() - 1].iter() {
+            fts_query.push('"');
+            fts_query.push_str(keyword);
+            fts_query.push('"');
+            fts_query.push(' ');
+        }
+        // The last keyword is special because we may want to add a `*` char to do a prefix match
+        // on it.  Do this if the input is > 3 characters, and there's no whitespace at the end.
+        if let Some(last_keyword) = keywords.last() {
+            let total_chars = keywords.iter().fold(0, |count, s| count + s.len());
+            let query_ends_in_whitespace =
+                matches!(self.keyword.chars().last(), Some(s) if s.is_whitespace());
+            if (total_chars > 3) && !query_ends_in_whitespace {
+                fts_query.push('"');
+                fts_query.push_str(last_keyword);
+                fts_query.push('"');
+                fts_query.push('*');
+            } else {
+                fts_query.push('"');
+                fts_query.push_str(last_keyword);
+                fts_query.push('"');
+            }
+        }
+        fts_query
```

Comment:
Can be simplified a bit. (I personally prefer raw string literals to avoid quote escaping)
```
        // Quote and append each term from `query`
        let keywords = self.parse_keywords();

        if keywords.is_empty() {
            return String::from(r#""""#);
        }

        let mut fts_query = keywords
            .iter()
            .map(|keyword| format!(r#""{keyword}""#))
            .collect::<Vec<_>>()
            .join(" ");


        // If the input is > 3 characters, and there's no whitespace at the end.
        // We want to append a `*` char to the end to do a prefix match on it.
        if !self.keyword.ends_with(' ') && keywords.iter().fold(0, |count, s| count + s.len()) > 3 {
            fts_query.push('*');
        }

        fts_query
```